std/compress/io.go

```diff
@@ -114,7 +114,7 @@ func UnpackIntoBytesHint(_ *big.Int, ins, outs []*big.Int) error {
 }
 
 // ReadNum reads the slice c as a big endian number in base radix
-func ReadNum(api frontend.API, c []frontend.Variable, radix int) frontend.Variable {
+func ReadNum(api frontend.API, c []frontend.Variable, radix frontend.Variable) frontend.Variable {
```

Comment:
why is the radix a variable now? is it really "variable" ??

Reply:
Just to cover possible cases where the radix is large(r than 64 bits)

Reply:
then use a big.Int here ? It conveys more the "this is a constant" part IMO.